macros/parsers/parserLinearRelation.pl

```diff
@@ -0,0 +1,386 @@
+	$context->{name}                       = "LinearRelation";
+	$context->{precedence}{LinearRelation} = $context->{precedence}{special};
+	$context->{value}{LinearRelation}      = "LinearRelation";
+	$context->{value}{Formula}             = "LinearRelation::formula";
+	$context->parens->remove("<");
+	$context->flags->set(standardForm => 0);
+	$context->operators->add(
+		'=' => {
+			precedence    => .5,
+			associativity => 'left',
+			type          => 'bin',
+			string        => ' = ',
+			kind          => 'eq',
+			eval          => sub { $_[0] == $_[1] },
+			class         => 'LinearRelation::inequality',
+			formulaClass  => "LinearRelation"
+		},
+		'<' => {
+			precedence    => .5,
+			associativity => 'left',
+			type          => 'bin',
+			string        => ' < ',
+			kind          => 'lt',
+			eval          => sub { $_[0] < $_[1] },
+			class         => 'LinearRelation::inequality',
+			formulaClass  => "LinearRelation"
+		},
+		'>' => {
+			precedence    => .5,
+			associativity => 'left',
+			type          => 'bin',
+			string        => ' > ',
+			kind          => 'gt',
+			reverse       => 'lt',
+			eval          => sub { $_[0] > $_[1] },
+			class         => 'LinearRelation::inequality',
+			formulaClass  => "LinearRelation"
+		},
+		'<=' => {
+			precedence    => .5,
+			associativity => 'left',
+			type          => 'bin',
+			string        => ' <= ',
+			TeX           => '\le ',
+			kind          => "le",
+			eval          => sub { $_[0] <= $_[1] },
+			class         => 'LinearRelation::inequality',
+			formulaClass  => "LinearRelation",
+			alternatives  => [ '=<', "\x{2264}" ]
+		},
+		'>=' => {
+			precedence    => .5,
+			associativity => 'left',
+			type          => 'bin',
+			string        => ' >= ',
+			TeX           => '\ge ',
+			kind          => 'ge',
+			reverse       => 'le',
+			eval          => sub { $_[0] >= $_[1] },
+			class         => 'LinearRelation::inequality',
+			formulaClass  => "LinearRelation",
+			alternatives  => [ '=>', "\x{2265}" ]
+		},
+		'!=' => {
+			precedence    => .5,
+			associativity => 'left',
+			type          => 'bin',
+			string        => ' != ',
+			TeX           => '\ne ',
+			kind          => 'ne',
+			eval          => sub { $_[0] != $_[1] },
+			class         => 'LinearRelation::inequality',
+			formulaClass  => "LinearRelation",
+			alternatives  => [ '<>', "\x{2260}" ]
+		},
+	);
+	main::PG_restricted_eval('sub LinearRelation {Value->Package("LinearRelation()")->new(@_)}');
+}
+
+#    $R = LinearRelation("x + y + 2z <= 5");
+#    $R = Formula("x + y + 2z <= 5");
+#    $R = Compute("x + y + 2z <= 5");
+#    $R = LinearRelation([1,2,1], [1,1,2], "<=");
+#    $R = LinearRelation([1,2,1], 5, "<=");
+
+sub new {
+	my ($self, $N, $p, $bop) = @_;
+	my $class   = ref($self) || $self;
+	my $context = (Value::isContext($_[0]) ? shift : $self->context);
+	my $formula = "Value::Formula";
+	return shift if scalar(@_) == 1 && ref($_[0]) eq 'LinearRelation';
```

Comment:
Something looks wrong here.  I think you should put this back to the way that you had it before @pstaabp and my comments on unpacking arguments on the first line.  We probably should have kept our mouths shut on this for now.  With the current setup of MathObjects that just doesn't work, and @dpvc should be heeded.  Lines 191 and 193 will not work right with this setup because `$self` has not been shifted out of the input argument array `@_`.